### PR TITLE
React1/Week2

### PR DESCRIPTION
**Pull request by Shweta-MG**

### CODE REVIEW

File: react/react1/week2/todo-list/src/Components/Home.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            description: desc,
+            deadline: date,
+            id: id,
+        }
+        console.log(newToDo);
+        setTodos([...todos, newToDo])
+    }
+
+
+    const [todos, setTodos] = useState([
```
Comment (LucasIversen) [inline react/react1/week2/todo-list/src/Components/Home.js:27]: I'll always put my states in the top of my components, but that's only a small detail

File: react/react1/week2/todo-list/src/Components/Home.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import React , { useState } from "react";
+import ToDoList from "./ToDo/ToDoList";
+import ToDoForm from "./ToDo/ToDoForm";
+
+
+const Home = () => {
+
+    const onDeleteHandler = (id) => {
+        const newToDos = todos.filter(todo => todo.id !== id);
```
Comment (LucasIversen) [inline react/react1/week2/todo-list/src/Components/Home.js:9]: really nice

File: react/react1/week2/todo-list/src/Components/Home.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        setTodos(newToDos);
+    }
+
+    const addToDo = (desc, date) => {
+        let id;       
+        todos.length !== 0 ? id = todos[todos.length - 1].id + 1 : id = 0;
```
Comment (LucasIversen) [inline react/react1/week2/todo-list/src/Components/Home.js:15]: Nice!

File: react/react1/week2/todo-list/src/Components/Home.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+        const newToDo = {
+            description: desc,
+            deadline: date,
+            id: id,
+        }
+        console.log(newToDo);
+        setTodos([...todos, newToDo])
```
Comment (LucasIversen) [inline react/react1/week2/todo-list/src/Components/Home.js:23]: Good use of dot operator

File: react/react1/week2/todo-list/src/Components/Home.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+
+   
+    return (  
+        <div className="home">
+            <ToDoForm addToDo={addToDo} />
+            <ToDoList
+                todos={todos}
+                title={'All are the To-Dos'}
+                deleteHandler={onDeleteHandler} />
```
Comment (LucasIversen) [inline react/react1/week2/todo-list/src/Components/Home.js:49]: Yes!! Parsing a function as a prop is exactly the way we'd make the sub component!

File: react/react1/week2/todo-list/src/Components/ToDo/ToDoForm.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        e.preventDefault();        
+        addToDo(desc, date)
+        //console.log(newToDo);
+
+        setDate('');
+        setDesc('');
+    }
+
+    return (  
+        <form className="form" onSubmit={submitHandler}>
```
Comment (LucasIversen) [inline react/react1/week2/todo-list/src/Components/ToDo/ToDoForm.js:20]: Good form!

File: react/react1/week2/todo-list/src/Components/ToDo/ToDoForm.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
+import { useState } from 'react';
+import './ToDoForm.css';
+
+const ToDoForm = (props) => {
+    const addToDo = props.addToDo;
+    const [desc, setDesc] = useState('');
```
Comment (LucasIversen) [inline react/react1/week2/todo-list/src/Components/ToDo/ToDoForm.js:6]: We could have also made a single input state

File: react/react1/week2/todo-list/src/Components/ToDo/ToDoList.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
+import './ToDoList.css';
+
+const ToDoList = (props) => {
+    const todos = props.todos;
+    const title = props.title;
+    const deleteHandler = props.deleteHandler;
```
Comment (LucasIversen) [inline react/react1/week2/todo-list/src/Components/ToDo/ToDoList.js:6]: Remember how we used object destructuring:

```javascript
const {todos, title, deleteHandler} = props;
```

File: react/react1/week2/todo-list/src/Components/ToDo/ToDoList.js [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+   
+    return (  
+        <div className="todo-list">
+            <h2>{title}</h2>
+            {todos.length === 0 ? 'No To-Dos to display' : todos.map((todo) => (<div className="todo-preview" key={todo.id}>
+                <h2> {todo.description}</h2>
+                <p> {todo.deadline}</p>  
+                <button onClick={() => {deleteHandler(todo.id)}}> Delete </button>
```
Comment (LucasIversen) [inline react/react1/week2/todo-list/src/Components/ToDo/ToDoList.js:14]: Really nice!

**Review by LucasIversen — APPROVED** [review]
Hi Shweta. Really good job! It seems that you have really caught on the the react way, so keep it up 💯

File: react/react1/week2/todo-list/src/Components/ToDo/ToDoList.js (hunk 8 above)
Comment (Shweta-MG) [reply react/react1/week2/todo-list/src/Components/ToDo/ToDoList.js:6]: Yeah!! I was just thinking to document this option too. Just for reference in future. :)

File: react/react1/week2/todo-list/src/Components/Home.js (hunk 1 above)
Comment (Shweta-MG) [reply react/react1/week2/todo-list/src/Components/Home.js:27]: I was expecting the comment, but let it be like that. I will follow this advice in future tutorials. :)

**Comment by Shweta-MG** [comment]
Hi @LucasIversen  Thank you very much for your time and guidance. I have made note of comments and will follow them in future tutorials. I thank you again. :)

**Pull request merged**